peekingduck/pipeline/nodes/model/jde_mot/jde_files/models.py

```diff
@@ -0,0 +1,375 @@
+    """YOLO layer"""
+
+    def __init__(
+        self,
+        anchors: List[Tuple[float, float]],
+        nC: int,
+        nID: int,
+        nE: int,
+        img_size: Tuple[int, int],
+        yolo_layer: int,
+        device: torch.device,
+    ) -> None:
+        super(YOLOLayer, self).__init__()
+        self.layer = yolo_layer
+        nA = len(anchors)
+        self.anchors = torch.FloatTensor(anchors)
+        self.nA = nA  # number of anchors (3)
+        self.nC = nC  # number of classes (80)
+        self.nID = nID  # number of identities
+        self.img_size = 0
+        self.emb_dim = nE
+        self.shift = [1, 3, 5]
+        self.device = device
+
+        self.SmoothL1Loss = nn.SmoothL1Loss()
+        self.SoftmaxLoss = nn.CrossEntropyLoss(ignore_index=-1)
+        self.CrossEntropyLoss = nn.CrossEntropyLoss()
+        self.IDLoss = nn.CrossEntropyLoss(ignore_index=-1)
+        self.s_c = nn.Parameter(-4.15 * torch.ones(1))  # -4.15
+        self.s_r = nn.Parameter(-4.85 * torch.ones(1))  # -4.85
+        self.s_id = nn.Parameter(-2.3 * torch.ones(1))  # -2.3
+
+        self.emb_scale = math.sqrt(2) * math.log(self.nID - 1) if self.nID > 1 else 1
+
+    def forward(
+        self,
+        p_cat: torch.Tensor,
+        img_size: Tuple[int, int],
+    ) -> torch.Tensor:
+        """Defines the computation performed at every call.
+
+        Args:
+            p_cat (torch.Tensor): Input from the previous layer.
+            img_size (Tuple[int, int]): Size of image frame.
+
+        Returns:
+            (torch.Tensor): YOLO prediction tensor.
+        """
+        p, p_emb = p_cat[:, :24, ...], p_cat[:, 24:, ...]
+        nB, nGh, nGw = p.shape[0], p.shape[-2], p.shape[-1]
+
+        if self.img_size != img_size:
+            create_grids(self, img_size, nGh, nGw)
+
+            if p.is_cuda:
+                self.grid_xy = self.grid_xy.cuda()
+                self.anchor_wh = self.anchor_wh.cuda()
+
+        p = (
+            p.view(nB, self.nA, self.nC + 5, nGh, nGw)
+            .permute(0, 1, 3, 4, 2)
+            .contiguous()
+        )  # prediction
+
+        p_emb = p_emb.permute(0, 2, 3, 1).contiguous()
+        p_box = p[..., :4]
+        p_conf = p[..., 4:6].permute(0, 4, 1, 2, 3)  # Conf
+
+        p_conf = torch.softmax(p_conf, dim=1)[:, 1, ...].unsqueeze(-1)
+        p_emb = F.normalize(
+            p_emb.unsqueeze(1).repeat(1, self.nA, 1, 1, 1).contiguous(), dim=-1
+        )
+        p_cls = torch.zeros(nB, self.nA, nGh, nGw, 1).to(self.device)  # Temp
+        p = torch.cat([p_box, p_conf, p_cls, p_emb], dim=-1)
+        p[..., :4] = decode_delta_map(p[..., :4], self.anchor_vec.to(p), self.device)
+        p[..., :4] *= self.stride
+
+        return p.view(nB, -1, p.shape[-1])
+
+
+class Darknet(nn.Module):
+    """YOLOv3 object detection model"""
+
+    def __init__(
+        self, cfg_dict: Dict[str, Any], nID: int = 0, test_emb: bool = False
+    ) -> None:
+        super(Darknet, self).__init__()
```

Comment:
Check if removing arguments in super breaks anything, remove if nothing breaks

Reply:
Removing arguments in super as nothing breaks otherwise.